spec/requests/api/v1/user/logout_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'DELETE api/v1/users/sign_out', type: :request do
+  describe 'DELETE session' do
+    let(:user) { create(:user) }
+
+    it 'logout a user' do
+      # initial sign in to generate a token and response
+      post user_session_path, params: {
+        email: user.email,
+        password: user.password
+      }, as: :json
+
+      expect(user.reload.tokens.count).to eq 1
+
+      # sign out request using header values from sign in response
+      delete destroy_user_session_path, params: {
+        'access-token': response.header['access-token'],
+        client: response.header['client'],
+        uid: response.header['uid']
+      }, as: :json
+
+      expect(json['errors']).to be_blank
+      expect(response).to be_ok
+      # user token should be deleted following sign out
+      expect(user.reload.tokens.count).to eq 0
```

Comment:
I'd recommend having a helper to generate a session instead of using the sign-in endpoint:
```
def auth_headers
  user.create_new_auth_token
end
```
Then, you'd do something like:
`delete destroy_user_session_path, headers: auth_headers, as: :json`

Also, there should be a case when the user is not logged in.

Reply:
the suggest was applied in the next commit, thanks